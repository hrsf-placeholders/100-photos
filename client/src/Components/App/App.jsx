import React from 'react';
import Styled from './AppStyles.js';
import axios from 'axios';
import Gallery from '../Gallery/Gallery.jsx';
import dummyData from '../../sampleData.js';

class App extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      data: [],
      numOfImages: 0,
      header: ''
    }
  }

  componentDidMount() {
    // var randomId = Math.floor(Math.random() * 100) + 1; // => use this to dynamically render
    var setId = 1;
    axios.get(`http://localhost:3002/api/photos/${setId}`)

      .then((response) => {
        var photos = response.data;
        var relativeHeader = '';

        var length = photos.length;
        if (length > 1) {
          relativeHeader = length + ' Photos';
        } else if (length === 1) {
          relativeHeader = '1 Photo';
        } else if (length === 0) {
          relativeHeader = '0 Photos'
        }

        let newState = {
          data: photos,
          numOfImages: photos.length,
          header: relativeHeader
        }

        this.setState(newState);
      })
      .catch(function (error) {
        console.log(error);
      });
  }

  render() {
    console.log("state", this.state);
    return (
      <Styled.PhotoGallery>
        <Styled.Header>{this.state.header}</Styled.Header>

        <Gallery data={this.state.data} imgTotal={this.state.numOfImages} />

      </Styled.PhotoGallery>
    )
  }
}

export default App;
