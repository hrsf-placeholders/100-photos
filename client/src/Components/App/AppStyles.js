import styled from 'styled-components';

//minimum w/h => 330 X 280
const PhotoGallery = styled.div`
  border: 1px solid black; 
  padding: 5px;
  justify-content: center;
  position: static;
  max-height: 100%; 
  max-width: 613px;
  min-height: 100%; 
  min-width: 352px;
  `;

const Header = styled.h2`
  font-size: 24px;
  font-weight: 700;
  line-height: 32px;
  font-family: BrandonText,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
`;

export default { PhotoGallery, Header }