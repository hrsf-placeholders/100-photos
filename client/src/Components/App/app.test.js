import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import App from './App.jsx';
import {PhotoGallery, Header} from './AppStyles.js';
import Adapter from 'enzyme-adapter-react-16';
import dummyData from '../../sampleData.js';

Enzyme.configure({ adapter: new Adapter() });

describe("App component", () => {
  const wrapper = shallow(<App/>);
  test("renders", () => {
    expect(wrapper.exists()).toBe(true);
  })
  it("snapshot test", () => {
    expect(wrapper).toMatchSnapshot()
  })
  test("data in state present after mounting", () => {
    wrapper.setState({ data: dummyData })
    expect(wrapper.state('data')).toBeTruthy();
  })
});