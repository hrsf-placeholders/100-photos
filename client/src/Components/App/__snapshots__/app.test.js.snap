// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot test 1`] = `
<AppStyles__PhotoGallery>
  <AppStyles__Header>
    0 Photos
  </AppStyles__Header>
  <Gallery
    data={Array []}
    imgTotal={0}
  />
</AppStyles__PhotoGallery>
`;
